symbol.md

```diff
@@ -0,0 +1,181 @@
+  #=> Hail Lightening Larry
+
+```
+
+What if we need a collection of just the names of the soldiers, we could using the map function do
+
+```ruby
+  soldiers.map{ |soldier|  soldier.send(:name) } #=> ["Big Mike", "Small John", "Lightening Larry"]
+
+  # Using our proc armory
+  proc = Proc.new { |soldier| soldier.send(:name) }
+  soldiers.map(&proc) #=> ["Big Mike", "Small John", "Lightening Larry"]
+```
+
+With symbol#proc, we can simplify this further
+
+```ruby
+  soldiers.map(&:name) #=> ["Big Mike", "Small John", "Lightening Larry"]
+```
+
+Now you know why we needed to make those stops we made at symbols and procs as such the question now won't be what each components of the code mean but rather, **How they work together?**
+
+The **&** calls `to_proc` on the object, and passes it as a block to the method. As such, if you put **&** in front of a Proc instance in the argument position, it will be interpreted as a block. If you combine something other than a Proc instance with **&**, then implicit class casting will try to convert that to a Proc instance using `to_proc` method defined on that object (if any). In the case of a Symbol instance, like above, `to_proc` works in this way:
+
+```ruby
+  class Symbol
+    def to_proc
+      Proc.new { |x| x.send(self) }
+    end
+  end
+```
+What the `to_proc` implementation in the **Symbol** class does is to call a method with the same name as the `symbol` (i.e. self) on the passed in object. So say the symbol name is `:attack` and the passed in argument is `enemy`, the result will be `enemy.send('attack')` which is equivalent to `enemy.attack`
+
+So for each of the soldiers in the array, the map method passes it to the proc as `x`. Therefore, the argument to map is interpreted as
+```ruby
+  soldiers.map { |x| x.name } # where x in this case is each soldier in the array
+```
+Note that this `Symbol#proc` conversion is valid for all methods that accept a proc.
+
+With that knowledge, to return the soldiers weapon, we can easily do
+
+```ruby
+  soldiers.map(&:weapon) #=> ["Bazooka", "Pistol", "Uzi"]
+
+  # to return the soldiers salaries
+  soldiers.collect(&:salary) #=> [4000, 5000, 8000]
+
+  #to get soldiers with grade_a_weapon
+  soldiers.select(&:grade_a_weapon?) #=> [#<Context::Soldier:0x000000017065b0 @name="Big Mike", @weapon="Bazooka">, #<Context::Soldier:0x00000001706538 @name="Small John", @weapon="Pistol">]
+  soldiers.select(&:grade_a_weapon?).map(&:weapon) #=> ["Bazooka", "Pistol"]
+```
+_Here's one for you to try out: write code to retrieve the last_name of soldiers with grade_a_weapon. Share your implementation in the comment section._
+
+Now let's match towards a more ambitious adventure.
+
+We can override the `to_proc` method in the Symbol class to achieve custom result. For instance say we want to always append the team captain's name to all `to_proc` requests;
+
+```ruby
+  class Symbol
+    def to_proc
+      Proc.new { |x| "#{x.send(self)} (signed John Doe Commander)" }
```

Comment:
Inform the readers of the dark side of doing this. They should never do it. This is for the purpose of learning.